gdzig/class/ClassDB.mixin.zig

```diff
@@ -0,0 +1,1139 @@
+        @ptrCast(class_name),
+        @ptrCast(subgroup_name),
+        @ptrCast(prefix),
+    );
+}
+
+/// Registers an integer constant on an extension class in the ClassDB.
+///
+/// @since 4.1
+pub inline fn registerIntegerConstant(class_name: *const StringName, enum_name: *const StringName, constant_name: *const StringName, constant_value: i64, is_bitfield: bool) void {
+    raw.classdbRegisterExtensionClassIntegerConstant(
+        raw.library,
+        @ptrCast(class_name),
+        @ptrCast(enum_name),
+        @ptrCast(constant_name),
+        constant_value,
+        @intFromBool(is_bitfield),
+    );
+}
+
+/// Virtual method info for registration.
+pub const VirtualMethodInfo = struct {
+    name: *const StringName,
+    flags: MethodFlags = .{},
+    return_value: PropertyInfo,
+    return_value_metadata: MethodArgumentMetadata = .none,
+    arguments: []const PropertyInfo = &.{},
+    arguments_metadata: []const MethodArgumentMetadata = &.{},
+};
+
+/// Registers a virtual method on an extension class in the ClassDB.
+/// This allows scripts or other extensions to implement the method.
+///
+/// @since 4.3
+pub inline fn registerVirtualMethod(class_name: *const StringName, info: VirtualMethodInfo) void {
+    raw.classdbRegisterExtensionClassVirtualMethod(
+        raw.library,
+        @ptrCast(class_name),
+        &c.GDExtensionClassVirtualMethodInfo{
+            .name = @ptrCast(@constCast(info.name)),
+            .method_flags = @bitCast(info.flags),
+            .return_value = @bitCast(info.return_value),
+            .return_value_metadata = @intFromEnum(info.return_value_metadata),
+            .argument_count = @intCast(info.arguments.len),
+            .arguments = @ptrCast(@constCast(info.arguments.ptr)),
+            .arguments_metadata = @ptrCast(@constCast(info.arguments_metadata.ptr)),
+        },
+    );
+}
+
+/// Unregisters an extension class in the ClassDB.
+///
+/// @since 4.1
+pub inline fn unregisterClass(class_name: *const StringName) void {
+    raw.classdbUnregisterExtensionClass(
+        raw.library,
+        @ptrCast(class_name),
+    );
+}
+
+// Imports needed inside the mixin (before @mixin stop)
+const godot = @import("../gdzig.zig");
+const object_module = godot.object;
+const Allocator = std.mem.Allocator;
+const Child = std.meta.Child;
+const CallError = godot.CallError;
+const PropertyError = godot.PropertyError;
+const RID = @import("../builtin.zig").RID;
+const MethodFlags = @import("../global.zig").MethodFlags;
+const PropertyHint = @import("../global.zig").PropertyHint;
+const PropertyUsageFlags = @import("../global.zig").PropertyUsageFlags;
+
+// @mixin stop
+
+const std = @import("std");
+
+const c = @import("gdextension");
+const raw = &@import("../gdzig.zig").raw;
+
+const Object = @import("../class.zig").Object;
+const String = @import("../builtin.zig").String;
+const StringName = @import("../builtin.zig").StringName;
+const Variant = @import("../builtin.zig").Variant;
```

Comment:
Some docs here be nice. Even if it was just "How to update this for a new version". In here be dragons.